tooling/maven/camel-package-maven-plugin/src/main/java/org/apache/camel/maven/packaging/PackageDataFormatMojo.java

```diff
@@ -515,7 +515,7 @@ private static String schemaSubDirectory(String javaType) {
     private Optional<JavaClassSource> doParseJavaClassSource(String className) {
         try {
             Path srcDir = project.getBasedir().toPath().resolve("src/main/java");
-            Path file = srcDir.resolve(className.replace('.', '/') + ".java");
+            Path file = srcDir.resolve(className.replaceAll("<.*>", "").replace('.', '/') + ".java");
```

Comment:
Great catch. I wonder if you could add a code comment above this why you are doing this reg exp replaceAll - as when we look at this code in 6 month or later then we wouldnt remember why.